tests/console/yast2_cmdline.pm

```diff
@@ -40,8 +42,22 @@ sub run_yast_cli_test {
 }
 
 sub run {
+    my $cmd;
+
     select_console 'root-console';
 
+    if (is_sle) {
+        if (sle_version_at_least('15') and get_var('REPO_SLE15_MODULE_BASESYSTEM_SOURCE')) {
+            $cmd = "ar -f " . "$utils::OPENQA_FTP_URL/" . get_var('REPO_SLE15_MODULE_BASESYSTEM_SOURCE') . " repo-source";
+        }
+        elsif (sle_version_at_least('12-SP4') and get_var('REPO_SLES_SOURCE')) {
+            $cmd = "ar -f " . "$utils::OPENQA_FTP_URL/" . get_var('REPO_SLES_SOURCE') . " repo-source";
+        }
+    }
```

Comment:
there is no else to this `is (is_sle)` so in the case of openSUSE the variable `$cmd` stays unitialized so further down the "zypper ar" call is skipped but you still call `ref`. I do not think this should be necessary this way

Reply:
Yes, I would prefer to leave openSUSE modifications for now. We need to solve how to sync source repos to o3 first. Afterwards we can get back to this issue.

Reply:
Yes, the repo syncing part is still unsolved. This here is more about the general programming best practice to always think about the else-path of an if. You could simply move the zypper call into the if or add an else with die.